js/interface.js

```diff
@@ -2371,37 +2383,54 @@ $('[name="fl-store-keywords"]').on('tokenfield:createtoken', function (e) {
 $('.redirectToSettings, [data-change-settings]').on('click', function (event) {
   event.preventDefault();
 
-  Fliplet.Studio.emit('close-overlay', {
-    name: 'publish-apple'
-  });
+  saveProgressOnClose().then(function () {
+    Fliplet.Studio.emit('close-overlay', {
+      name: 'publish-apple'
+    });
+  
+    Fliplet.Studio.emit('overlay', {
+      name: 'app-settings',
+      options: {
+        size: 'large',
+        title: 'App Settings',
+        section: 'appSettingsGeneral',
+        appId: Fliplet.Env.get('appId')
+      }
+    });
+  })
```

Comment:
Move `.catch()` to this line

```suggestion
  }).catch(function (err) {
```